cmd/explaintest/t/index_join.test

```diff
@@ -10,3 +10,13 @@ analyze table t1, t2;
 -- we expect the following two SQL chose t2 as the outer table
 explain select /*+ TIDB_INLJ(t1, t2) */ * from t1 join t2 on t1.a=t2.a;
 explain select * from t1 join t2 on t1.a=t2.a;
+
+-- Test https://github.com/pingcap/tidb/issues/10516
+drop table if exists t1, t2;
+create table t1(a int not null, b int not null);
+create table t2(a int not null, b int not null, key a(a));
+
+set @@tidb_opt_insubq_to_join_and_agg=0;
+explain select /*+ TIDB_INLJ(t2) */ * from t1 where t1.a in (select t2.a from t2);
+show warnings;
+set @@tidb_opt_insubq_to_join_and_agg=1;
```

Comment:
How about adding one more test to enable `tidb_opt_insubq_to_join_and_agg` and specify `t1` in `TIDB_INLJ` hint?
```
mysql> explain select /*+ TIDB_INLJ(t1) */ * from t1 where t1.a in (select t2.a from t2);
+----------------------------+----------+------+-------------------------------------------------------------------------------------------------+
| id                         | count    | task | operator info                                                                                   |
+----------------------------+----------+------+-------------------------------------------------------------------------------------------------+
| Projection_8               | 10000.00 | root | test.t1.a, test.t1.b                                                                            |
| └─IndexJoin_12             | 10000.00 | root | inner join, inner:IndexLookUp_11, outer key:test.t2.a, inner key:test.t1.a                      |
|   ├─IndexLookUp_11         | 10.00    | root |                                                                                                 |
|   │ ├─IndexScan_9          | 10.00    | cop  | table:t1, index:a, range: decided by [eq(test.t1.a, test.t2.a)], keep order:false, stats:pseudo |
|   │ └─TableScan_10         | 10.00    | cop  | table:t1, keep order:false, stats:pseudo                                                        |
|   └─StreamAgg_24           | 8000.00  | root | group by:col_1, funcs:firstrow(col_0)                                                           |
|     └─IndexReader_25       | 8000.00  | root | index:StreamAgg_16                                                                              |
|       └─StreamAgg_16       | 8000.00  | cop  | group by:test.t2.a, funcs:firstrow(test.t2.a)                                                   |
|         └─IndexScan_23     | 10000.00 | cop  | table:t2, index:a, range:[NULL,+inf], keep order:true, stats:pseudo                             |
+----------------------------+----------+------+-------------------------------------------------------------------------------------------------+
```